l10n_be_cooperator_national_number/models/res_partner.py

```diff
@@ -35,18 +35,22 @@ def get_be_national_register_number(self):
     def update_belgian_national_number(self, national_number):
         self.ensure_one()
         result = None
-        if national_number:
-            existing = self.get_be_national_register_number_id_number()
-            # Update
-            if existing:
+        existing = self.get_be_national_register_number_id_number()
+        # Update
+        if existing:
+            if not national_number:
+                # TODO: should we check against require_national_number before
+                # unlinking?
+                existing.unlink()
```

Comment:
should this use sudo()?

Reply:
so this is typically run by the user managing subscription.requests, except in the portal module. let's just call sudo _there_.